package/cfg/rke2-cis-1.24-hardened/policies.yaml

```diff
@@ -0,0 +1,339 @@
+
+      - id: 5.1.5
+        text: "Ensure that default service accounts are not actively used. (Automated)"
+        audit: check_for_default_sa.sh
+        tests:
+          test_items:
+            - flag: "true"
+              compare:
+                op: eq
+                value: "true"
+              set: true
+        remediation: |
+          Create explicit service accounts wherever a Kubernetes workload requires specific access
+          to the Kubernetes API server.
+          Modify the configuration of each default service account to include this value
+          automountServiceAccountToken: false
+        scored: true
+
+      - id: 5.1.6
+        text: "Ensure that Service Account Tokens are only mounted where necessary (Manual)"
+        type: "manual"
+        remediation: |
+          Modify the definition of pods and service accounts which do not need to mount service
+          account tokens to disable it.
+        scored: false
+
+      - id: 5.1.7
+        text: "Avoid use of system:masters group (Manual)"
+        type: "manual"
+        remediation: |
+          Remove the system:masters group from all users in the cluster.
+        scored: false
+
+      - id: 5.1.8
+        text: "Limit use of the Bind, Impersonate and Escalate permissions in the Kubernetes cluster (Manual)"
+        type: "manual"
+        remediation: |
+          Where possible, remove the impersonate, bind and escalate rights from subjects.
+        scored: false
+
+  - id: 5.2
+    text: "Pod Security Standards"
+    checks:
+      - id: 5.2.1
+        text: "Ensure that the cluster has at least one active policy control mechanism in place (Manual)"
+        type: "manual"
+        remediation: |
+          Ensure that either Pod Security Admission or an external policy control system is in place
+          for every namespace which contains user workloads.
+        scored: false
+
+      - id: 5.2.2
+        text: "Minimize the admission of privileged containers (Automated)"
+        audit: kubectl get psp global-restricted-psp && kubectl get psp global-restricted-psp -o json | jq -r ".spec.runAsUser.rule" || kubectl get psp restricted-noroot-psp && kubectl get psp restricted-noroot-psp -o json | jq -r ".spec.runAsUser.rule"
+        tests:
+           test_items:
+             - flag: "MustRunAsNonRoot"
+               compare:
+                 op: eq
+                 value: "MustRunAsNonRoot"
+               set: true
+        remediation: |
+          Add policies to each namespace in the cluster which has user workloads to restrict the
+          admission of privileged containers.
+        scored: true
+
+      - id: 5.2.3
+        text: "Minimize the admission of containers wishing to share the host process ID namespace (Automated)"
+        audit: "kubectl get psp -o json | jq .items[] | jq -r 'select((.spec.hostPID == null) or (.spec.hostPID == false))' | jq .metadata.name | wc -l | xargs -I {} echo '--count={}'"
```

Comment:
These are no longer valid checks for RKE2 v1.25+, which have moved to PSAs. CIS-1.24 was AFAIK supposed to be changed to help validate only v1.25+ versions of both k3s and rke2.

Reply:
@dereknola so should we keep these checks as manual checks([as it is in kube-bench](https://github.com/aquasecurity/kube-bench/blob/main/cfg/cis-1.24/policies.yaml))?
cc: @andypitcher @JonCrowther

Reply:
I think on CIS 1.24, these checks will be present and from CIS 1.25 these will be removed. 
@andypitcher

Reply:
Sure. We can convert them to manual for now.

Reply:
@dereknola No actually we can't convert them to manual, the current behaviour (pass on K8s 1.24 cluster and fail on 1.25 cluster) because CIS 1.24 profiles are supposed to be run on K8s 1.24 clusters and as 1.24 supports PSP these checks will pass. 
On K8s 1.25 cluster these checks won't work as expected because PSP was removed. So when CIS 1.25 Profiles will be released, these checks will be marked as manual because CIS 1.25 profiles can't be run on K8s 1.24 cluster.
cc: @andypitcher

Reply:
FYI @andypitcher is working on adding CIS 1.25 profiles. When that's released, CIS 1.24 profiles will no longer be used to scan K8s 1.25 clusters.